src/main/java/htsjdk/samtools/read/Read.java

```diff
@@ -0,0 +1,1438 @@
+
+    /**
+     * Gets the character value associated with the tag.
+     *
+     * <p>Equivalent to an attribute with <b>TYPE A</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation gets the attribute with {@link #getAttribute(String)}, checking if it is a {@link Character}.
+     *
+     * @param tag the attribute tag.
+     *
+     * @return the character value.
+     *
+     * @throws SAMException if the tag is invalid following the contract of {@link ReadAttribute#isValidTag(String)}.
+     * @throws SAMException if the attribute value is not a character.
+     */
+    default Optional<Character> getCharacterAttribute(final String tag) {
+        final Optional<Object> attr = getAttribute(tag);
+        if (!attr.isPresent()) {
+            return Optional.empty();
+        }
+        final Object val = attr.get();
+
+        if (val instanceof Character) {
+            return Optional.of((Character)val);
+        }
+        throw new SAMException("Value for tag " + tag + " is not a Character: " + val.getClass());
+    }
+
+    /**
+     * Gets the singed integer value associated with the tag.
+     *
+     * <p>Equivalent to an attribute with <b>TYPE i</b> or any <b>TYPE B</b> integer (signed or unsigned) in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation gets the attribute with {@link #getAttribute(String)}, checking if it is a numeric value within the integer range.
+     *
+     * @param tag the attribute tag.
+     *
+     * @return the integer value.
+     *
+     * @throws SAMException if the tag is invalid following the contract of {@link ReadAttribute#isValidTag(String)}.
+     * @throws SAMException if the attribute value is not an integer.
+     */
+    default Optional<Integer> getIntegerAttribute(final String tag) {
+        final Optional<Object> attr = getAttribute(tag);
+        if (!attr.isPresent()) {
+            return Optional.empty();
+        }
+        final Object val = attr.get();
+
+        if (!(val instanceof Number)) {
+            throw new SAMException("Value for tag " + tag + " is not Number: " + val.getClass());
+        }
+        final long longVal = ((Number)val).longValue();
+        if (longVal < Integer.MIN_VALUE || longVal > Integer.MAX_VALUE) {
+            throw new SAMException("Value for tag " + tag + " is not in Integer range: " + longVal);
+        }
+        return Optional.of((int) longVal);
+    }
+
+    /**
+     * Gets the float value associated with the tag.
+     *
+     * <p>Equivalent to an attribute with <b>TYPE f</b> or <b>TYPE B</b> float in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation gets the attribute with {@link #getAttribute(String)}, checking if it is a {@link Float}.
+     *
+     * @param tag the attribute tag.
```

Comment:
(two character string)

Reply:
Changed everywhere (but already specified in the `SAMException` documentation).